crypto/finite_field.ts

```diff
@@ -195,5 +195,12 @@ function createField(p: bigint, t: bigint, twoadicRoot: bigint) {
       }
       return BigInt('0b' + binary.reverse().join(''));
     },
+    leftShift(x: bigint, bits: number, maxBitSize: number = 64) {
+      let shifted = x << BigInt(bits);
+      return BigInt('0b' + shifted.toString(2).slice(-maxBitSize));
```

Comment:
nit: more efficient would be
```suggestion
      return shifted & ((1n << BigInt(maxBitSize)) - 1n);
```